src/KnownProperties.jl

```diff
@@ -0,0 +1,67 @@
+########################################################################
+# This is to introduce a common language to ask whether or not a given
+# property (which is potentially hard to compute) is known for some
+# object `x`. 
+########################################################################
+
+@doc raw"""
+    is_known(x::Any, f::Function)
+
+Return whether a property of an object `x` called for by `f(x)` is known. 
+
+Note: The default implementation throws an error. It is the programmer's 
+responsibility to implement appropriate methods for their individual 
+types and properties. See `src/KnownProperties.jl` for details.
+"""
+function is_known(x::Any, f::Function)
+  # If the object `x` has attributes and an attribute with the name 
+  # of `f` happens to be stored, return it in good faith.
+  hasfield(typeof(x), :__attrs) && has_attribute(x, nameof(f)) && return true
+  # Otherwise notify the user that no method is implemented rather than 
+  # just saying "no". The explicit call to this method suggests that 
+  # the callee expects the function to return something useful and 
+  # not having a working method at hand is probably a bug. 
+  error("no method implemented to check whether property $(nameof(f)) is known for object $x")
+end
+
+@doc raw"""
+    is_known(x::Any, f::Function, args...; kwargs...)
+
+Return whether the property of `x` called for by `f(x, args...)` is known. 
+
+Note: The default implementation throws an error. It is the programmer's 
+responsibility to implement appropriate methods for their individual 
+types and properties. See `src/KnownProperties.jl` for details.
+"""
+function is_known(x::Any, f::Function, args...; kwargs...)
+  error("no method implemented to check whether property $(nameof(f)) with arguments $(args) and keyword arguments $(kwargs) is known for object $x")
+end
+
+
+# In general, it is the programmer's responsibility to implement 
+# a method for their types and properties. The above already implements 
+# a generic deflection to the attributes where applicable. But this 
+# will by far not catch all cases. 
+# 
+# We give a few samples. 
+
+is_known(i::Int, ::typeof(is_even)) = true # Almost no computation cost, so known. 
+is_known(i::Int, ::typeof(is_odd)) = true # Almost no computation cost, so known. 
+
+# The dimension of a ring is difficult to compute in general. However, for 
+# some rings it is easy.
+is_known(R::MPolyRing{<:FieldElem}, ::typeof(dim)) = true
```

Comment:
This looks more like examples that should go into the documentation? (Maybe the docstring?)

Reply:
If you think the docstring is a good place for this, I'm happy to put it there.

Reply:
The docstring which does `AbstractAlgebra.is_known(5, is_even)` would indeed be a natural place for the `is_known(i::Int, ::typeof(is_even)) = true` example (BTW ditch the `i` it is redundant).

We don't need the `is_odd` example.

The `is_known(R::MPolyRing{<:FieldElem}, ::typeof(dim))` method looks genuinely useful, as in: `dim` is one of the function we want to support by `is_known`. So I'd keep it (but remove the useless `R`). Of course we can argue *where* it should be. I am totally open about that.